libraries/extensions/ros2-bridge/python/src/typed/serialize.rs

```diff
@@ -31,81 +34,154 @@ impl serde::Serialize for TypedValue<'_> {
     {
         match &self.type_info.data_type {
             DataType::UInt8 => {
-                let uint_array: UInt8Array = self.value.clone().into();
-                let number = uint_array.value(0);
-                serializer.serialize_u8(number)
+                let uint_array: &UInt8Array = self.value.as_primitive();
+                if uint_array.len() == 1 {
+                    let number = uint_array.value(0);
+                    serializer.serialize_u8(number)
+                } else {
+                    let mut s = serializer.serialize_seq(Some(uint_array.len()))?;
+                    for value in uint_array.iter() {
+                        s.serialize_element(&value)?;
+                    }
+                    s.end()
+                }
```

Comment:
The `DataType::Uint8` specifies that the ROS2 message expects a single `u8` value. So serializing multiple values will result in invalid data when it's read by the receiver.

Instead of ignoring the additional values, we should probably return an error if `uint_array.len() > 1`.

(same for the other basic data types below)

Reply:
So, the thing is that if you have a specification like:
```
uint8[3] uint8_values
```
This would result in a `ListArray` of `Uint8Arrays` where in the list there is 1 element of an array with 3 elements. When we parse it with the serializer and we go through the list and we call serialize_element on the `UInt8Arrays` we need to parse the 3 elements. Just as within the discussion: https://github.com/dora-rs/dora-autoware/discussions/9#discussioncomment-8133366

and 
```
uint8 uint8_value
```
This would result in a `Uint8Arrays` where in the list there is 1 element. This is going to be parsed as a single element with no sequence just as before.

`UInt8` is not only for scalar and maybe we should add our own scalar type to make sure that we have a clear spec. 

The main benefit of doing the serializing the way this PR do it is that we are aligned with current dora-rs arrays.